Python/Include/Autobox.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef ROVER_PYTHON_AUTOBOX_HPP
+#define ROVER_PYTHON_AUTOBOX_HPP
+#include <memory>
+#include <type_traits>
+#include <pybind11/pybind11.h>
+#include "Rover/Box.hpp"
+#include "Rover/Constant.hpp"
+#include "Rover/Evaluator.hpp"
+#include "Rover/Noncopyable.hpp"
+
+namespace Rover::Details {
+  template<typename T>
+  class PythonBox : private Noncopyable {
+    public:
+      using Type = T;
+
+      template<typename ObjFwd, std::enable_if_t<!std::is_convertible_v<ObjFwd,
+        PythonBox>>* = nullptr>
```

Comment:
Why should I copy objects? You mentioned it was a heavy operation.

Reply:
Use ```std::move``` to avoid the copy.

Reply:
I mean, say I have PythonBox constructor defined as follows:
PythonBox(object obj) : m_obj(std::move(obj)) {}
Then, I have user code as follows:
// case 1: OK, no copies, 2 moves
auto box = Box(PythonBox(std::move(obj));
// case 2: 1 copy and 1 move instead of 2 copies of a pointer (pass-by-reference):
auto box = Box(PythonBox(obj));
Not saying that the latter case is very frequent, but it is not impossible, is it?

Reply:
On the other hand, PythonBox and PythonRange are not parts of the API (being under Details) and std::move is the their only use case. Is that the reason?

Reply:
python_autobox is actually a part of the API, and using it as python_autobox<object>(subgenerator) will inflict a copy due to the PythonBox constuctor in case it doesn't work with lvalue references.